docsite/source/error-handling.html.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Error Handling
+layout: gem-single
+name: dry-operation
+---
+
+When using dry-operation, errors are handled through the `Failure` type from [dry-monads](/gems/dry-monads/). Each step in your operation should return either a `Success` or `Failure` result. When a step returns a `Failure`, the operation short-circuits, skipping the remaining steps and returning the failure immediately.
+
+You'll usually handle the failure from the call site, where you can pattern match on the result to handle success and failure cases. However, sometimes it's useful to encapsulate some error handling logic within the operation itself.
+
+### Global error handling
+
+You can define a global failure handler by implementing an `#on_failure` method in your operation class:
+
+```ruby
+class CreateUser < Dry::Operation
+  def initialize(logger:)
+    @logger = logger
+  end
+
+  def call(input)
+    attrs = step validate(input)
+    user = step persist(attrs)
+    step notify(user)
+    user
+  end
+
+  private
```

Comment:
I added these `private` lines here, since I think we want to encourage this method not leaking into the objects' public APIs. Does that sound OK to you?

Reply:
Makes total sense 👍